### PR TITLE
Add CES survey for search product, order, customer

### LINKED ISSUE #6307: Customer Effort Score: include additional features
**Description:** we want to start triggering the Customer Effort in certain features that weren't included in the initial launch:

1. Import Products
2. Product Categories
3. Product Attributes
4. Product Tags
5. Search (orders, products, customers)

### PR DESCRIPTION
**Pull request by ilyasfoo**
This PR adds CES survey trigger on search for products, orders, and customers. Note that the survey action is shared between them, which means that it will only trigger once for either.

We're also adding an event prop `search_area` which can be either `product`, `shop_order`, or `customer`.

Partially fixes https://github.com/woocommerce/woocommerce-admin/issues/6307.

### Detailed test instructions:

- Make sure tracking is enabled in settings: 
```
/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com
```
- Delete the option `woocommerce_ces_shown_for_actions` to make sure CES prompt triggers when updating settings.
- Enable the logging of Tracks events to your browser dev console:
```
localStorage.setItem( 'debug', 'wc-admin:tracks' );
```

**Testing search on products:**
- Go to Products > All Products.
- Type in anything in search bar, click on "Search products".
- Observe CES prompt "How easy was it to use search?" is displayed.

**Testing search on orders:**
- Go to Orders > Orders.
- Type in anything in search bar, click on "Search orders".
- Observe CES prompt "How easy was it to use search?" is displayed.

**Testing search on customers:**
- Go to Customers.
- Type in anything in search bar, and press enter.
- Observe CES prompt "How easy was it to use search?" is displayed.

### CODE REVIEW

File: src/Features/CustomerEffortScoreTracks.php [hunk 1]
```diff
@@ -234,4 +269,27 @@ public function run_on_update_options() {
...
+		$allowed_types = array( 'product', 'shop_order' );
+
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		// We're only interested for certain post types.
+		if ( empty( $_GET['post_type'] ) || ! in_array( $_GET['post_type'], $allowed_types, true ) ) {
+			return;
+		}
+
+		// Determine whether request is search by "s" GET parameter.
+		if ( empty( $_GET['s'] ) ) {
```
Comment (adrianduffell) [inline src/Features/CustomerEffortScoreTracks.php:286]: Could [`get_search_query()`](https://developer.wordpress.org/reference/functions/get_search_query/) be used here?

File: src/Features/CustomerEffortScoreTracks.php [hunk 2]
```diff
@@ -234,4 +269,27 @@ public function run_on_update_options() {
...
+
+	/**
+	 * Determine on initiating CES survey on searching for product or orders.
+	 */
+	public function run_on_load_edit_php() {
+		$allowed_types = array( 'product', 'shop_order' );
+
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		// We're only interested for certain post types.
+		if ( empty( $_GET['post_type'] ) || ! in_array( $_GET['post_type'], $allowed_types, true ) ) {
```
Comment (adrianduffell) [inline src/Features/CustomerEffortScoreTracks.php:281]: Could [`get_current_screen()`](https://developer.wordpress.org/reference/functions/get_current_screen/) be used here to find the current post type?

**Review by adrianduffell — CHANGES_REQUESTED** [review]
Thanks @ilyasfoo this tested well and LGTM overall 👍 

I just left some suggestions to use native WP functions so we could avoid disabling PHPCS.

**Comment by timmyc** [comment]
Looks like this will be good to merge after a few tweaks...

File: src/Features/CustomerEffortScoreTracks.php (hunk 1 above)
Comment (ilyasfoo) [reply src/Features/CustomerEffortScoreTracks.php]: I've tried this, but unfortunately `get_search_query()` returns empty string during our hook call. Bummer 😞

File: src/Features/CustomerEffortScoreTracks.php (hunk 2 above)
Comment (ilyasfoo) [reply src/Features/CustomerEffortScoreTracks.php]: Made the change in d1fb96b. Thanks for the suggestion @adrianduffell!

**Review by adrianduffell — APPROVED** [review]
LGTM 🚀 

Thanks for switching over to the native WP functions where it was possible!

It would have been good to have some tests for this. Perhaps that's something we could add later in cooldown.

**Pull request merged**